### PR TITLE
test: add product switch to storybook

### PR DESCRIPTION
**Pull request by stefanoScalzo**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/1954

#### Please provide a brief summary of this pull request.
This pr introduces product switch to storybook. It has an accessibility issue due to the button which will be fixed in another PR
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by stefanoScalzo** [comment]
blocked by https://github.com/SAP/fundamental-ngx/pull/2477/files

File: src/stories/product-switch/product-switch.stories.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                [triggers]="triggersVar">
+            <fd-product-switch-body
+                [forceListMode]="forceListModeVar"
+                [products]="objectVar"
+                [dragAndDropEnabled]="dragAndDropEnabled">
+            </fd-product-switch-body>
+        </fd-product-switch>
+    `,
+    props: {
+        forceListModeVar: boolean('Mobile Mode', false),
```
Comment (rengare) [inline src/stories/product-switch/product-switch.stories.ts:43]: avoid `Var` suffix it's not needed

File: src/stories/product-switch/product-switch.stories.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        closeOnEscapeKey: boolean('Close On Esc', false),
```
Comment (rengare) [inline src/stories/product-switch/product-switch.stories.ts:44]: The correct name would be `onCloseKeyPressed`
in general, when you deal with events `on` prefix should be added

**Comment by droshev** [comment]
@stefanoScalzo the pipelines are failing. Once they are green i can approve it

**Comment by stefanoScalzo** [comment]
> The pipelines are failing
> I noticed that the product switch doesn't work:
> 
> * when we set close on esc -> it doesn't close - can you log an issue?
> * focus trap doesn't work - can you log an issue?
> * not Hide Arrow doesn't work - can you log an issue?
> * triggers on hover doesn't seem to be working -  can you double check this?
> * isOpen doesn't work - can you log an issue?
> 
> Improvement - can you render the button in the middle of the screen so we can play with the placement?

Ill change where the button is rendered but for the first part, they do work you need to play with it a bit more because stroy book will lag sometimes

**Comment by stefanoScalzo** [comment]
> I tried all below and nothing worked
> I noticed that the product switch doesn't work:
> 
> * when we set close on esc -> it doesn't close - can you log an issue?
> * focus trap doesn't work - can you log an issue?
> * not Hide Arrow doesn't work - can you log an issue?
> * triggers on hover doesn't seem to be working - can you double check this?
> * isOpen doesn't work - can you log an issue?

see the comment above. They do work, storybook takes time loading you need to fiddle with the ui

**Pull request closed without merging**